rfcs/006-package-management.md

```diff
@@ -0,0 +1,357 @@
+`fetchGit`. There was some discussion
+[here](https://github.com/tweag/nickel/issues/329#issuecomment-967372858)
+on the advantages and disadvantages of inline imports.
+
+### Alternative: toml manifest
+
+Maybe the manifest should be in some plain-data format like toml. This would
+be easier to modify programmatically, and it would prevent people from putting
+lots of complicated code in their manifest files.
+The disadvantage of a plain-data format is that you don't get a nice contract
+for it and you can't use ADTs for the dependencies.
+
+We discussed this point in office hours, and the general sentiment was that
+it's ok to allow the manifest to be interpreted. If someone wants to use
+that power to create a ridiculously complicated manifest, that's their problem.
+
+### Alternative: shorthand for registry imports
+
+Cargo allows a shorthand like
+
+```toml
+    "github/tweag/foo" = "1.2.0"
+```
+
+instead of
+
+```nickel
+    foo = 'Index { package = "github/tweag/foo", version = "1.2.0" }
+```
+
+Since we expect registry imports to be the common case, maybe it's worth having
+a shorthand?
+
+### Alternative: manifest file name
+
+Bikeshed the name "electroplate.ncl."
+
+## Import statements
+
+The manifest file assigns a name to each dependency; to import the dependency
+named `foo` you simply write `import foo`. That is, an `import` statement either
+takes a string in quotes -- in which case it imports a path -- or an identifier
+without quote -- in which case it imports a package.
+
+The `import foo` expression evaluates to the contents of `main.ncl` in `foo`'s
+root directory.
+
+### Alternative: other entry points
+
+We've hardcoded `main.ncl` as the entry point of every package, but what if
+they want to expose multiple entry points? For example, node allows a package's
+manifest to specify the entry point(s). This is probably not very important to
+support, as you can just put
+
+```nickel
+{
+  other = import "./other.ncl",
+  blah = import "./blah.ncl",
+}
+```
+
+in your package's `main.ncl`, to provide "other" and "blah" as other entry points.
+
+### Alternative: entry points are top-level files
+
+Instead of hardcoding `main.ncl`, we could say that every file in the package's
+top-level directory is publicly accessible. Package authors could keep implementation
+details private by putting code in subdirectories.
+
+### Alternative: namespace the import tool
+
+Our initial intention for packaging was to allow for the usage of multiple
+different package management tools. This RFC only proposes one such tool, but
+maybe the import syntax could be designed with other tools in mind. For example,
+it could be `import foo from electroplate` with the idea that future nickel
+versions might add, say, `import foo from nix-flake`.
+
+## CLI interface
+
+We will add a new CLI tool (called `plate`) that wraps the nickel CLI and
+adds package management. It will offer a superset of the nickel CLI's commands
```

Comment:
This seems to be a huge argument in favor of embeding package management in nickel's CLI.

At some point, I can hardly imagine anybody _not_ using package management. That would mean everybody would use `plate` instead of `nickel` anyways.